src/Blocks/custom/carousel/assets/index.js

```diff
@@ -0,0 +1,26 @@
+import domReady from '@wordpress/dom-ready';
+
+domReady(() => {
+	const selector = '.js-block-carousel';
+	const elements = document.querySelectorAll(selector);
+
+	if (elements.length) {
+		const eventName = new CustomEvent('carouselInit');
+
+		import('./carousel-slider').then(({ CarouselSlider }) => {
+			[...elements].forEach((element) => {
+				const carouselSlider = new CarouselSlider({
+					element,
+					blockClass: 'block-carousel',
```

Comment:
we dont use bem as a js selector. and you must have js prefix

Reply:
why is this key named block class? can this Js functionality be used on anything other than blocks?

Reply:
I guess it could be used in a Carousel component, but except that it's pretty specific to this block.

Reply:
ok but use js prefix then

Reply:
That's the name from the eightshift carousel block.

I guess I can change it to just `class` because it could be used in a Carousel component, but otherwise it's pretty specific to this block/component.

Reply:
check you attributes array on output, you have js selector there to use